src/types/redis_json.cc

```diff
@@ -102,14 +108,18 @@ rocksdb::Status Json::Get(const std::string &user_key, const std::vector<std::st
   if (paths.empty()) {
     res = std::move(json_val);
   } else if (paths.size() == 1) {
-    auto get_res = json_val.Get(paths[0]);
+    auto json_path_result = JsonPath::BuildJsonPath(paths[0]);
+    if (!json_path_result) return rocksdb::Status::InvalidArgument(json_path_result.Msg());
+    auto get_res = json_val.Get(json_path_result.GetValue());
     if (!get_res) return rocksdb::Status::InvalidArgument(get_res.Msg());
     res = *std::move(get_res);
   } else {
     for (const auto &path : paths) {
-      auto get_res = json_val.Get(path);
+      auto json_path_result = JsonPath::BuildJsonPath(path);
+      if (!json_path_result) return rocksdb::Status::InvalidArgument(json_path_result.Msg());
+      auto get_res = json_val.Get(json_path_result.GetValue());
       if (!get_res) return rocksdb::Status::InvalidArgument(get_res.Msg());
```

Comment:
I think maybe there's no difference between the original code and this patch here.

Reply:
It has the ability to handle the legacy path, isn't it?

Reply:
```c++
  StatusOr<JsonValue> Get(const JsonPath &json_path) const {
    try {
      return json_path.EvalQueryExpression(value);
    } catch (const jsoncons::jsonpath::jsonpath_error &e) {
      return {Status::NotOK, e.what()};
    }
  }
```

It will go to here.

Reply:
Oh I see, seems I can just `Get(std::string_view)` here.